Basic universe management APIs and GUI

PardhavMaradani opened the pull request:
* Base class for MDAnalysis visualization (`MDAVis`)
* APIs for Universe addition/listing/retrieval/deletion
* Subscriptable and iterable universes for easy data access
* GUI for universe listing and management (addition/deletion) and Trajectory info
* Example notebook with basic universe management calls
* Tests for all the new functionality

BradyAJohnston commented:
I'll have a more detailed look at this later this evening, but I'm happy to add `MDAnalysisTests` to one of the optional dependencies. The old tests that use current data should also be moved over to use trajectories from there as well probably, but something for another project.

yuxuanzhuang commented:
I will also take a look sometime this week.

Quick thought: there are a lot of `print`, should a logging mechanism be introduced to MN?

yuxuanzhuang commented on molecularnodes/mda/base.py:
```diff
@@ -0,0 +1,105 @@
+        self.universes = Universes(self._universes)
+        utils.subscribe_to_active_object()  # subscribe to active object changes
+
+    def add_universe(
```
If you add the same universe twice, you will get two separate MN universes but they are both linked back to the same mda universe? Is that something we want?

yuxuanzhuang commented on molecularnodes/mda/universes.py:
```diff
@@ -0,0 +1,92 @@
+from ..entities.trajectory import Trajectory
+
+
+class Universe:
```
The naming is getting really confusing now...maybe we should make them more explicit on what they actually are.

PardhavMaradani replied:
Yes, this could get confusing. The universes here are the ones in MN/Blender, which will have methods to add styles, annotations, etc later. Should we change them to something like `MNUniverse` or `BUniverse`? Do you guys have any recommendations on naming this?

PardhavMaradani replied:
Could you guys please review the refactored and renamed code and let me know if this still is an issue? I will leave this unresolved for now.

PardhavMaradani commented:
> I will also take a look sometime this week.
> 
> Quick thought: there are a lot of `print`, should a logging mechanism be introduced to MN?

We could use `loguru` like in `ggmolvis`. Brady, adding a dependency seems to be more than adding to the `dependencies` list in `pyproject.toml` due to `wheels` that I don't fully understand yet. Could you please help.

PardhavMaradani commented:
I refactored the universe classes and also renamed them as `UniverseManager` and `UniverseView` to be more explicit of what they do. I am open to renaming them again if needed.

PardhavMaradani commented:
I thought we were starting out with the objective to provide a clear and carefully crafted API interface along with an associated GUI for MDAnalysis users who would need better visualization capabilities. But if that is not the case and we're leaving that to other visualization tools like `ggmolvis` etc we could definitely expose `MNSession` and the `Trajectory` entity directly. To me, it seems a bit awkward to expose these directly but I agree that `MNSession` is already holding the entity details and `Trajectory` is what is already responsible for the view. I don't quite see the relevance of the `MDAnalysis` named GUI tab in the n-panel in this model.

Given that there is already a way to add trajectories (by instantiating a `Trajectory` object followed by a `create_object`), this isn't really a blocker for me. If you want to have APIs like `add_trajectory`, `get_trajectory` etc to `MNSession`, could you please help define clearly what these would look like (along with if any GUI support for them is needed and how they'd exactly look like). Given that this is how all users would interact with this, it would greatly help if you guys can outline this based on your experience and looking at it from the longer run. I could then help with implementing those as you guys see fit. For now, I could start working on the more independent aspects like the style branch support, annotations, etc.

BradyAJohnston commented:
I think go forward with implementing around the existing `Trajectory`. Other particulars that don't need to be exposed we could hide. There is certainly lots overall that could be improved & cleaned up on the `Trajectory` class.
